easybuild/easyblocks/c/cppcheck.py

```diff
@@ -0,0 +1,141 @@
+##
+# Copyright 2016 Forschungszentrum Juelich GmbH
+#
+# This file is part of EasyBuild,
+# originally created by the HPC team of Ghent University (http://ugent.be/hpc/en),
+# with support of Ghent University (http://ugent.be/hpc),
+# the Flemish Supercomputer Centre (VSC) (https://www.vscentrum.be),
+# Flemish Research Foundation (FWO) (http://www.fwo.be/en)
+# and the Department of Economy, Science and Innovation (EWI) (http://www.ewi-vlaanderen.be/en).
+#
+# http://github.com/hpcugent/easybuild
+#
+# EasyBuild is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation v2.
+#
+# EasyBuild is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with EasyBuild.  If not, see <http://www.gnu.org/licenses/>.
+##
+"""
+@author: Damian Alvarez (Forschungszentrum Juelich GmbH)
+"""
+import os
+import shutil
+
+from easybuild.easyblocks.generic.configuremake import ConfigureMake
+from easybuild.framework.easyconfig import CUSTOM
+from easybuild.tools.build_log import EasyBuildError
+from easybuild.tools.run import run_cmd
+
+
+class EB_cppcheck(ConfigureMake):
+    """
+    EasyBlock to install cppcheck
+    """
+    
+    @staticmethod
+    def extra_options(extra_vars=None):
+        """
+        Define if we are using rules or not, and if we are building the GUI
+        """
+        extra = {
+            'have_rules': [False, "Use rules", CUSTOM],
+            'build_gui': [False, "Build GUI", CUSTOM],
+        }
+        if extra_vars is None:
+            extra_vars = {}
+        extra.update(extra_vars)
+        return ConfigureMake.extra_options(extra_vars=extra)
+
+    def configure_step(self):
+        """
+        Run qmake on the GUI, if necessary
+        """
+        if self.cfg['build_gui']:
+            if self.cfg['have_rules']:
+                cmd = 'qmake HAVE_RULES=yes'
+            else:
+                cmd = 'qmake'
+            
+            try:
+                gui_dir = os.path.join(self.cfg['start_dir'], 'gui')
```

Comment:
move this line outside of the `try` to, so it's definitely defined in the `except` scope